tests/net/lib/http_server/crime/src/main.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2023, Emna Rekik
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/ztest.h>
+
+static const unsigned char inc_file[] = {
+#include "index.html.gz.inc"
+};
+
+static const unsigned char compressed_inc_file[] = {
+	0x1f, 0x8b, 0x08, 0x00, 0x00, 0x00, 0x00, 0x00, 0x02, 0xff, 0x35, 0x8e, 0xc1, 0x0a, 0xc2,
+	0x30, 0x0c, 0x86, 0xef, 0x7d, 0x8a, 0xec, 0x05, 0x2c, 0xbb, 0x87, 0x5c, 0x54, 0xf0, 0xe0,
+	0x50, 0x58, 0x41, 0x3c, 0x4e, 0x17, 0x69, 0x21, 0xa5, 0x65, 0x2d, 0x42, 0xdf, 0xde, 0xba,
+	0x6e, 0x21, 0x10, 0xf8, 0xf9, 0xbe, 0x9f, 0x60, 0x77, 0xba, 0x1d, 0xcd, 0xf3, 0x7e, 0x06,
+	0x9b, 0xbd, 0x90, 0xc2, 0xfd, 0xf0, 0x34, 0x93, 0x82, 0x3a, 0x98, 0x5d, 0x16, 0xa6, 0xa1,
+	0xc0, 0xe8, 0x7c, 0x14, 0x86, 0x91, 0x97, 0x2f, 0x2f, 0xa8, 0x5b, 0xae, 0x50, 0x37, 0x16,
+	0x5f, 0x61, 0x2e, 0x9b, 0x62, 0x7b, 0x7a, 0xb0, 0xbc, 0x83, 0x67, 0xc8, 0x01, 0x7c, 0x81,
+	0xd4, 0xd4, 0xb4, 0xaa, 0x5d, 0x55, 0xfa, 0x8d, 0x8c, 0x64, 0xac, 0x4b, 0x50, 0x77, 0xda,
+	0xa1, 0x8b, 0x19, 0xae, 0xf0, 0x71, 0xc2, 0x07, 0xd4, 0xf1, 0xdf, 0xdf, 0x8a, 0xab, 0xb4,
+	0xbe, 0xf6, 0x03, 0xea, 0x2d, 0x11, 0x5c, 0xb2, 0x00,
+};
+
+ZTEST(framework_tests_crime, test_gen_gz_inc_file)
+{
+	const char *data;
+	size_t len;
+
+	data = inc_file;
+	len = sizeof(inc_file);
+	len = len - 2;
+
+	zassert_equal(len, sizeof(compressed_inc_file), "Invalid compressed file size");
+
+	int result = memcmp(data, compressed_inc_file, len);
```

Comment:
I think the point of the HTTP server supporting CRiMe, is that we want the server to serve the compressed resource and we want to obtain the resource via an HTTP GET request.

This test does not exercise any HTTP server code paths at all, from what I can tell.

Reply:
The way that you would get the server to serve a specific resource is first by defining the service, and then defining the resource.